exes/mqtt-bridge/inc/config/publish_signals.hpp

```diff
@@ -25,12 +27,14 @@ template <>
 struct tfc::json::detail::to_json_schema<tfc::mqtt::config::signal_name> {
   template <auto Opts>
   static void op(auto& s, auto&) noexcept {
-    if (!s.oneOf.has_value()) {
-      s.oneOf = std::vector<tfc::json::detail::schematic>{};
-    }
-    for (auto const& signal : tfc::global::get_signals()) {
-      s.oneOf.value().push_back(tfc::json::detail::schematic{ .attributes{
-          tfc::json::schema{ .title = signal.name, .description = signal.description, .constant = signal.name } } });
+    s.oneOf = std::vector<schematic>{};
+    for (auto const& signal : global::get_signals()) {
+      // clang-format off
+      PRAGMA_GCC_WARNING_OFF(-Wmaybe-uninitialized)
+      // clang-format on
+      s.oneOf->emplace_back(schematic{
+          .attributes{ schema{ .title = signal.name, .description = signal.description, .constant = signal.name } } });
+      PRAGMA_GCC_WARNING_POP
```

Comment:
Instead of switching of these warnings you could create a temporary, populate that and then std::move into s.oneOf. This is also not the correct usage of this pragma should be `PRAGMA_GCC_WARNING_PUSH_OFF` to match your pop below.

Reply:
removed the warning and added a temp variable that is moved.